pkg/api/kubeconfig/generate_handler.go

```diff
@@ -197,6 +197,35 @@ func (h *GenerateHandler) ensureSaAndSecret(namespace, name string) (*corev1.Ser
 		}
 	}
 
+	secretName := sa.Name + "-token"
+	secretNamespace := sa.Namespace
+	secret, err := h.secretCache.Get(secretNamespace, secretName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return nil, nil, err
+	}
+	if apierrors.IsNotFound(err) {
+		secret, err = h.secretClient.Create(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      secretName,
+				Namespace: secretNamespace,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: rbacv1.SchemeGroupVersion.Version,
+						Kind:       rbacv1.ServiceAccountKind,
+						Name:       sa.Name,
+						UID:        sa.UID,
+					},
+				},
+				Annotations: map[string]string{
+					"kubernetes.io/service-account.name": sa.Name,
```

Comment:
for the key prfer to use `v1.ServiceAccountNameKey` instead and add `v1.ServiceAccountUIDKey` as well.

Reply:
updated